src/services/auth.ts

```diff
@@ -175,35 +187,40 @@ class AuthService {
         oidcUser = null;
       }
     }
-    let user: User | null = null;
     if (oidcUser) {
-      const { name, email, sub } = oidcUser.profile;
-      if (name && email && sub) {
+      const { sub, name, email } = oidcUser.profile;
+      if (sub && name && email && oidcUser.access_token) {
         const expired = oidcUser.expired ?? true;
         let fullName = name;
         user = {
           expired,
           name,
           fullName: name,
+          loginState: loginState.identified,
           email,
           extId: sub,
         };
         try {
-          const response = await fetchJson(new URL(sub, USERS_URL));
+          let tokenHeader: HeadersInit = { "X-Authorization": "Bearer " + oidcUser.access_token }
+          const response = await fetchJson(new URL(sub, USERS_URL), "GET", null, tokenHeader);
           if (response.status === 200) {
             const userData = await response.json();
-            const { id, status, title } = userData;
+            const { id, name, title } = userData;
             if (title) fullName = `${title} ${fullName}`;
+            if (name !== userData.name || email !== userData.email) user.loginState = loginState.needs_reregistration
+            else user.loginState = loginState.registered
```

Comment:
```suggestion
            const changed = name !== userData.name || email !== userData.email;
            const loginState = changed ? LoginState.needsReregistration : loginState.registered;
```

Reply:
Honestly, I think we should ditch the changed slot, since we now have the state that makes it redundant.